app/graphql/mutations/profile/associate_systems.rb

```diff
@@ -24,15 +24,51 @@ def resolve(args = {})
 
       def find_profile(profile_id)
         ::Pundit.authorize(
-          context[:current_user],
+          current_user,
           ::Profile.find(profile_id),
           :edit?
         )
       end
 
       def find_hosts(system_ids)
-        ::Pundit.policy_scope(context[:current_user], ::Host)
-                .where(id: system_ids)
+        existing_systems = ::Pundit.policy_scope(current_user, ::Host)
+                                   .where(id: system_ids)
+        save_hosts(system_ids - existing_systems.pluck(:id))
+        existing_systems
+      end
+
+      def save_hosts(ids)
+        ids.map do |id|
+          save_host(id)
+        end
+      end
+
+      def save_host(id)
+        i_host = inventory_host(id)
+        host = ::Host.find_or_initialize_by(
```

Comment:
This `find` is only meant to happen to avoid a host being initialized twice (two concurrent mutations) or when would that happen?

Reply:
It's a safeguard that _shouldn't_ ever be hit, but there could be a race if the host is created between lines 47 and 48 in some other process.